src/cmake/thirdparty/SetupAxomThirdParty.cmake

```diff
@@ -11,27 +11,59 @@
 # UMPIRE
 #------------------------------------------------------------------------------
 if (UMPIRE_DIR)
-    include(cmake/thirdparty/FindUmpire.cmake)
-    blt_register_library( NAME      umpire
-                          INCLUDES  ${UMPIRE_INCLUDE_DIRS}
-                          LIBRARIES umpire
-                          TREAT_INCLUDES_AS_SYSTEM ON)
+    if (NOT EXISTS "${UMPIRE_DIR}")
```

Comment:
I am wondering if we could change this to an axom `macro` (probably too restrictive for a `blt` macro ). Just by looking at what we do for `umpire` and `raja` there is certainly a lot of repetition.  Having a macro may simplify this quite a bit and streamline the process of adding a TPL....